src/snowcli/cli/nativeapp/teardown_processor.py

```diff
@@ -0,0 +1,190 @@
+from pathlib import Path
+from textwrap import dedent
+from typing import Dict
+
+import typer
+from rich import print
+from snowcli.cli.nativeapp.constants import (
+    COMMENT_COL,
+    EXTERNAL_DISTRIBUTION,
+    INTERNAL_DISTRIBUTION,
+    OWNER_COL,
+    SPECIAL_COMMENT,
+)
+from snowcli.cli.nativeapp.exceptions import (
+    CouldNotDropApplicationPackageWithVersions,
+)
+from snowcli.cli.nativeapp.manager import (
+    NativeAppCommandProcessor,
+    NativeAppManager,
+    ensure_correct_owner,
+)
+from snowcli.exception import SnowflakeSQLExecutionError
+from snowflake.connector.cursor import DictCursor
+
+
+class NativeAppTeardownProcessor(NativeAppManager, NativeAppCommandProcessor):
+
+    needs_confirm: bool = True
```

Comment:
Should remove from class scope as `needs_confirmation` gets it passed in as a variable; this instance variable is hidden / aliased